File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/person/PersonName.java
package io.quarkus.it.mongodb.panache.person;

import java.util.Objects;

public class PersonName {
    public String lastname;

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;
        PersonName that = (PersonName) o;
        return Objects.equals(lastname, that.lastname);
    }

    @Override
    public int hashCode() {
        return Objects.hash(lastname);
    }
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/person/PersonName.kt
package io.quarkus.it.mongodb.panache.person

class PersonName {
    var lastname: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is PersonName) return false

        if (lastname != other.lastname) return false

        return true
    }

    override fun hashCode(): Int {
        return lastname?.hashCode() ?: 0
    }
}
